### ISSUE TITLE
Error creating sql replicated primary-secondary instances: Operation failed because another operation was already in progress

### ISSUE THREAD

**Issue by AlexBulankou** [GoogleCloudPlatform/k8s-config-connector#16]
**TLDR**: This is expected that secondary instance cannot be created before primary, but also expected that the controller will not attempt to create the instance, until primary creation is complete and not issue calls that will result in errors.

Try to create 2 SQL instances:
```yaml
apiVersion: sql.cnrm.cloud.google.com/v1alpha2
kind: SQLInstance
metadata:
  labels:
    label-one: "value-one"
  name: mysql-instance-sample-with-replication-primary
spec:
  databaseVersion: MYSQL_5_7
  region: us-central1
  settings:
    tier: db-f1-micro
    backupConfiguration:
        binaryLogEnabled: true
        enabled: true
        startTime: "18:00"
    ipConfiguration:
      requireSsl: true
    locationPreference:
        zone: us-central1-b
---
apiVersion: sql.cnrm.cloud.google.com/v1alpha2
kind: SQLInstance
metadata:
  labels:
    label-one: "value-one"
  name: mysql-instance-sample-with-replication-secondary
spec:
  databaseVersion: MYSQL_5_7
  region: us-central1
  masterInstanceName: mysql-instance-sample-with-replication-primary
  mysqlReplicaConfiguration:
    connectRetryInterval: 30
  settings:
    tier: db-f1-micro
    ipConfiguration:
      requireSsl: true
    locationPreference:
        zone: us-central1-c
```

The first instance is created successfully, the second instance is not created and in the event log I can see the following errors, first this one: `Could not read response as json for google api sqladmin-v1beta4: reading the error from the request is not implemented`, which seems to be common and then another one with more details:

```
 jsonPayload: {
  controller:  "sqlinstance-controller"   
  error:  "error syncing resource: Create call failed: googleapi: Error 409: Operation failed because another operation was already in progress., operationInProgress"   
  level:  "error"   
  logger:  "kubebuilder.controller"   
  msg:  "Reconciler error"   
  request:  "default/mysql-instance-sample-with-replication-secondary"   
  stacktrace:  "cnrm.googlesource.com/cnrm/vendor/github.com/go-logr/zapr.(*zapLogger).Error
	/go/src/cnrm.googlesource.com/cnrm/vendor/github.com/go-logr/zapr/zapr.go:128
cnrm.googlesource.com/cnrm/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/src/cnrm.googlesource.com/cnrm/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:217
cnrm.googlesource.com/cnrm/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/go/src/cnrm.googlesource.com/cnrm/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158
cnrm.googlesource.com/cnrm/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
	/go/src/cnrm.googlesource.com/cnrm/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
cnrm.googlesource.com/cnrm/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil
	/go/src/cnrm.googlesource.com/cnrm/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
cnrm.googlesource.com/cnrm/vendor/k8s.io/apimachinery/pkg/util/wait.Until
	/go/src/cnrm.googlesource.com/cnrm/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88"   
  ts:  1563976375.8204267   
 }
```
Note that eventually, with the subsequent reconciliation cycle, the second instance is created.
This is expected that secondary instance cannot be created before primary, but also expected that the controller will not attempt to create the instance, until primary creation is complete.

**Comment by kibbles-n-bytes** [comment]
Looks like `masterInstanceName` should actually be converted to a Kubernetes-style reference as well. We'll update the schema so that the secondary instance will look like this:

```yaml
apiVersion: sql.cnrm.cloud.google.com/v1alpha3
kind: SQLInstance
metadata:
  labels:
    label-one: "value-one"
  name: mysql-instance-sample-with-replication-secondary
spec:
  databaseVersion: MYSQL_5_7
  region: us-central1
  masterInstanceRef:
    name: mysql-instance-sample-with-replication-primary
  mysqlReplicaConfiguration:
    connectRetryInterval: 30
  settings:
    tier: db-f1-micro
    ipConfiguration:
      requireSsl: true
    locationPreference:
        zone: us-central1-c
```

This will allow us to internally poll the API server until the primary instance is ready, to not have to hit GCP repeatedly during a known unready state.

Will respond here once the change is in.

**Comment by AlexBulankou** [comment]
Fixed with 0.0.8.
